mlflow/projects/__init__.py

```diff
@@ -196,25 +181,26 @@ def _expand_uri(uri):
     return os.path.abspath(uri)
 
 
-def _fetch_project(uri, use_temp_cwd, version=None, git_username=None, git_password=None):
+def _is_local_path(uri):
+    return urllib.parse.urlparse(uri).scheme == ''
+
+
+def _fetch_project(uri, force_tempdir, version=None, git_username=None, git_password=None):
     """
     Fetch a project into a local directory, returning the path to the local project directory.
     """
-    # Separating the uri from the subdirectory requested.
     parsed_uri, subdirectory = _parse_subdirectory(uri)
-    dst_dir = _get_dest_dir(parsed_uri, use_temp_cwd)
+    use_temp_dst_dir = force_tempdir or not _is_local_path(uri)
+    dst_dir = tempfile.mkdtemp() if use_temp_dst_dir else urllib.parse.urlparse(uri).path
     eprint("=== Fetching project from %s into %s ===" % (uri, dst_dir))
```

Comment:
Maybe condition this print statement based on `if use_temp_dst_dir`? Otherwise it is basically a no-op.